src/Chirp.Infrastructure/ChirpRepository/AuthorRepository.cs

```diff
@@ -7,130 +7,161 @@
+            return 0; // make this into an exception
+        }
+
+        int cheepsCount = DbSetAuthor.Entry(authorEntity)
+                    .Collection(_author => _author.Cheeps)
+                    .Query()
+                    .Count();
+        return cheepsCount;
+    }
+
+    public async Task<Author?> GetAuthorByName(string name)
+    {
+        // FirstOrDefault returns null if no User is found.
+        var author = await SearchFor(_author => _author.User.Name == name).FirstOrDefaultAsync();
+
+        return author;
+    }
+
+    //GetCheepsByAuthor should be replaced with GetCheepsByAuthorId, we should search by id not name
     public async Task<Tuple<List<CheepDTO>, int>> GetCheepsByAuthor(string author, int offset, int limit)
     {
         // Helge has said we're to assume Author.Name are unique for now.
 
         int cheepsCount = 0;
 
-        var authorEntity = await GetAuthorByName(author);
+        var authorEntity = SearchFor(_author => _author.User.Name == author).FirstOrDefault();
 
         if (authorEntity is null)
         {
             return new Tuple<List<CheepDTO>, int>(new List<CheepDTO>(), cheepsCount);
         }
         else
         {
-            cheepsCount = DbSet.Entry(authorEntity)
+            cheepsCount = DbSetAuthor.Entry(authorEntity)
                     .Collection(_author => _author.Cheeps)
                     .Query().Count();
         }
 
-        List<CheepDTO> cheeps = DbSet.Entry(authorEntity)
+        List<CheepDTO> cheeps = DbSetAuthor.Entry(authorEntity)
                     .Collection(_author => _author.Cheeps)
                     .Query()
                     .OrderByDescending(_cheep => _cheep.TimeStamp)
                     .Skip(offset).Take(limit)
                     .Select(_cheep => new CheepDTO
                     (
-                        _cheep.Author.Name,
+                        _cheep.Author.User.Name,
                         _cheep.Text,
-                        _cheep.TimeStamp.ToString()
+                        _cheep.TimeStamp
                     ))
                     .ToList()
                     ?? new List<CheepDTO>();
 
         return new Tuple<List<CheepDTO>, int>(cheeps, cheepsCount);
     }
 
-    public async Task<Author?> GetAuthorById(int id)
+    public async Task<List<CheepDTO>> GetCheepsByAuthorId(int id, int offset, int limit)
     {
-        return await DbSet.FindAsync(id);
-    }
+        int cheepsCount = 0;
+        
+        var authorEntity = SearchFor(_author => _author.User.UserId == id).FirstOrDefault();
 
-    public async Task<Author?> GetAuthorByName(string name)
-    {
-        // FirstOrDefault returns null if no Author is found.
-        var author = await SearchFor(_author => _author.Name == name).FirstOrDefaultAsync();
+        if (authorEntity is null)
+        {
+            return new List<CheepDTO>();
+        }
 
-        return author;
+        List<CheepDTO> cheeps = DbSetAuthor.Entry(authorEntity)
```

Comment:
We should honestly just make this into a method for cleaner code lol

Reply:
Nicely spotted. This could be corrected in another branch.